src/shared/metadata/cgroup_path_resolver.cc

```diff
@@ -16,29 +16,39 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <absl/strings/str_replace.h>
+#include <linux/magic.h>
+#include <sys/vfs.h>
 #include <filesystem>
 #include <regex>
 #include <string>
 #include <vector>
 
-#include <absl/strings/str_replace.h>
-
 #include "src/common/fs/fs_wrapper.h"
 #include "src/shared/metadata/cgroup_path_resolver.h"
 
+DEFINE_bool(force_cgroup2_mode, true, "Flag to force assume cgroup2 fs for testing purposes");
+
 namespace px {
 namespace md {
 
-StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path) {
+StatusOr<std::string> CGroupBasePath(std::string_view sysfs_path, bool FLAGS_force_cgroup2_mode) {
   // Different hosts may mount different cgroup dirs. Try a couple for robustness.
   const std::vector<std::string> cgroup_dirs = {"cpu,cpuacct", "cpu", "pids"};
 
+  std::string cgv2_base_path = absl::StrCat(sysfs_path, "/cgroup");
+  struct statfs info;
+  auto fs_status = statfs(cgv2_base_path.c_str(), &info);
+
   for (const auto& cgroup_dir : cgroup_dirs) {
     std::filesystem::path dir;
 
     // Attempt assuming naming scheme #1.
     std::string base_path = absl::StrCat(sysfs_path, "/cgroup/", cgroup_dir);
-    if (fs::Exists(base_path)) {
+
+    if (((fs_status == 0) && (info.f_type == CGROUP2_SUPER_MAGIC)) || FLAGS_force_cgroup2_mode) {
```

Comment:
Continuing a previous comment, you can pull `(fs_status == 0) && (info.f_type == CGROUP2_SUPER_MAGIC)` out of the loop too, since it's static. Give it a variable name like `cgroupv2` and it'll make the code more readable too.

Reply:
In fact, the loop is only really required for cgroup v1, so we should pull out the cgroupv2 return value out of the loop too. We can return early if we detect cgroup v2 (without going into the loop).

Reply:
fixed in https://github.com/pixie-io/pixie/pull/635/commits/4d090b63538e2e74806b029362bdee46680520a2